### ISSUE TITLE
Problem syncing app sync failed without message

### ISSUE THREAD

**Issue by peterbosalliandercom** [argoproj/argo-cd#1887]
**Describe the bug**
We try to sync our project but it gives errors in the log without any clue what the error is. 

It used to be running fine, it is a large project but we never ran into problems like this.

When doing helm template on the project and apply them manually they are all synced fine. 

When doing a partial sync it works fine sometimes for some resources but others fail, it is very hard to find out why
Is there any pattern we should use to debug these issues?

Is there any way we can show the real error message?

**To Reproduce**
run : 
argocd app sync <application> 

in the log errors appear.

**Expected behavior**
Syncing of course ;)

**Screenshots**

**Version**
1.0.0

```shell
argocd-server: v1.0.0+c74ca22
  BuildDate: 2019-05-16T20:04:21Z
  GitCommit: c74ca22023cbee245cb83fb23224978fbe23703c
  GitTreeState: clean
  GoVersion: go1.11.4
  Compiler: gc
  Platform: linux/amd64
  Ksonnet Version: 0.13.1
```

**Logs**

this is a subset of the errors. the message is empty

partial sync
```
Helm Values:        values-prd-cluster.yaml
Sync Policy:        <none>
Sync Status:        OutOfSync from HEAD (921e418)
Health Status:      Healthy
Operation:          Sync
Sync Revision:      921e418f047f91740559c623f32c36c4b83ceb47
Phase:              Succeeded
Start:              2019-07-05 18:28:26 +0200 DST
Finished:           2019-07-05 18:28:32 +0200 DST
Duration:           6s
Message:            successfully synced (no manifests)
```
```
 level=info msg="added resource RoleBinding/forza-rolebinding status: SyncFailed, message: " application=forza-prd-gitops
time="2019-07-05T15:54:17Z" level=info msg="added resource RoleBinding/forza-rolebinding status: SyncFailed, message: " application=forza-prd-gitops
time="2019-07-05T15:54:18Z" level=info msg="added resource RoleBinding/ir-rolebinding status: SyncFailed, message: " application=forza-prd-gitops
time="2019-07-05T15:54:18Z" level=info msg="added resource RoleBinding/inpai-rolebinding status: SyncFailed, message: " application=forza-prd-gitops
time="2019-07-05T15:54:18Z" level=info msg="added resource RoleBinding/dag-rolebinding status: SyncFailed, message: " application=forza-prd-gitops
time="2019-07-05T15:54:20Z" level=info msg="added resource RoleBinding/ir-rolebinding status: SyncFailed, message: " application=forza-prd-gitops
time="2019-07-05T15:54:20Z" level=info msg="added resource RoleBinding/ir-rolebinding status: SyncFailed, message: " application=forza-prd-gitops
time="2019-07-05T15:54:21Z" level=info msg="added resource RoleBinding/forza-rolebinding status: SyncFailed, message: " application=forza-prd-gitops
time="2019-07-05T15:54:21Z" level=info msg="added resource RoleBinding/ant-rolebinding status: SyncFailed, message: " application=forza-prd-gitops
```

**Have you thought about contributing a fix yourself?**

Open Source software thrives with your contribution. It not only gives skills you might not be able to get in your day job, it also looks amazing on your resume. 

If you want to get involved, check out the 
[contributing guide](https://github.com/argoproj/argo-cd/blob/master/docs/CONTRIBUTING.md), then reach out to us on [Slack](https://argoproj.github.io/community/join-slack) so we can see how to get you started.

**Comment by peterbosalliandercom** [comment]
Another issue is logged with the same reason maybe: https://github.com/argoproj/argo-cd/issues/1884

**Comment by peterbosalliandercom** [comment]
Please help because we cannot continue, maybe rollback to previous version?

**Comment by peterbosalliandercom** [comment]
In the logs 

time="2019-07-08T15:43:34Z" level=error msg="`kubectl --kubeconfig /dev/shm/110241418 -f - apply -n kafka-acc --dry-run` failed: " execID=rl5Tt
time="2019-07-08T15:43:34Z" level=info msg="adding resource result, status: 'SyncFailed', phase: 'Failed', message: '`kubectl --kubeconfig /dev/shm/110241418 -f - apply -n kafka-acc --

**Comment by peterbosalliandercom** [comment]
when running this command in the container it works fine:
```
kubectl --kubeconfig /dev/shm/110241418 -f - apply -n kafka-acc --dry-run
```

**Comment by peterbosalliandercom** [comment]
more logs :

time="2019-07-08T16:02:59Z" level=error msg="`kubectl --kubeconfig /dev/shm/402707993 -f - apply -n test-ops --dry-run` failed: " execID=nqIC9
time="2019-07-08T16:02:59Z" level=info msg="adding resource result, status: 'SyncFailed', phase: 'Failed', message: '`kubectl --kubeconfig /dev/shm/402707993 -f - apply -n test-ops --dry-run` failed:'" application=forza-tst-gitops kind=Namespace name=test-int namespace=test-ops phase=Sync
time="2019-07-08T16:02:59Z" level=error msg="`kubectl --kubeconfig /dev/shm/529599140 -f - apply -n test-ops --dry-run` failed: " execID=Xco9c
time="2019-07-08T16:02:59Z" level=info msg="adding resource result, status: 'SyncFailed', phase: 'Failed', message: '`kubectl --kubeconfig /dev/shm/529599140 -f - apply -n test-ops --dry-run` failed:'" application=forza-tst-gitops kind=Namespace name=kafka-acc namespace=test-ops phase=Sync
time="2019-07-08T16:02:59Z" level=error msg="`kubectl --kubeconfig /dev/shm/160424141 -f - apply -n test-ops --dry-run` failed: " execID=vPcKA
time="2019-07-08T16:02:59Z" level=info msg="adding resource result, status: 'SyncFailed', phase: 'Failed', message: '`kubectl --kubeconfig /dev/shm/160424141 -f - apply -n test-ops --dry-run` failed:'" application=forza-tst-gitops kind=Namespace name=test-ops namespace=test-ops phase=Sync
time="2019-07-08T16:03:00Z" level=error msg="`kubectl --kubeconfig /dev/shm/749763874 -f - apply -n test-ops --dry-run` failed: " execID=quDou
time="2019-07-08T16:03:00Z" level=info msg="adding resource result, status: 'SyncFailed', phase: 'Failed', message: '`kubectl --kubeconfig /dev/shm/749763874 -f - apply -n test-ops --dry-run` failed:'" application=forza-tst-gitops kind=Namespace name=kube-public namespace=test-ops phase=Sync
time="2019-07-08T16:03:00Z" level=error msg="`kubectl --kubeconfig /dev/shm/910295823 -f - apply -n test-ops --dry-run` failed: " execID=RB5d8
time="2019-07-08T16:03:00Z" level=info msg="adding resource result, status: 'SyncFailed', phase: 'Failed', message: '`kubectl --kubeconfig /dev/shm/910295823 -f - apply -n test-ops --dry-run` failed:'" application=forza-tst-gitops kind=Namespace name=helm namespace=test-ops phase=Sync
time="2019-07-08T16:03:00Z" level=error msg="`kubectl --kubeconfig /dev/shm/971937338 -f - apply -n test-ops --dry-run` failed: " execID=ymbQI
time="2019-07-08T16:03:00Z" level=info msg="adding resource result, status: 'SyncFailed', phase: 'Failed', message: '`kubectl --kubeconfig /dev/shm/971937338 -f - apply -n test-ops --dry-run` failed:'" application=forza-tst-gitops kind=Namespace name=kafka-prd namespace=test-ops phase=Sync
time="2019-07-08T16:03:00Z" level=error msg="`kubectl --kubeconfig /dev/shm/943609368 -f - apply -n test-ops --dry-run` failed: " execID=cqXlb
time="2019-07-08T16:03:00Z" level=info msg="adding resource result, status: 'SyncFailed', phase: 'Failed', message: '`kubectl --kubeconfig /dev/shm/943609368 -f - apply -n test-ops --dry-run` failed:'" application=forza-tst-gitops kind=Namespace name=jenkins namespace=test-ops phase=Sync
time="2019-07-08T16:03:01Z" level=info msg="Applying resource StorageClass/st1-encrypted in cluster: https://k8s-api.test.appx.cloud, namespace: test-ops"
time="2019-07-08T16:03:01Z" level=info msg="Applying resource StorageClass/gp2-encrypted in cluster: https://k8s-api.test.appx.cloud, namespace: test-ops"
time="2019-07-08T16:03:01Z" level=info msg="Applying resource StorageClass/nfs in cluster: https://k8s-api.test.appx.cloud, namespace: test-ops"
time="2019-07-08T16:03:01Z" level=info msg="Applying resource StorageClass/gp2 in cluster: https://k8s-api.test.appx.cloud, namespace: test-ops"
time="2019-07-08T16:03:01Z" level=info msg="kubectl --kubeconfig /dev/shm/433537290 -f - apply -n test-ops --dry-run" dir= execID=h2TJq
time="2019-07-08T16:03:01Z" level=info msg="kubectl --kubeconfig /dev/shm/064936673 -f - apply -n test-ops --dry-run" dir= execID=BA0k1
time="2019-07-08T16:03:01Z" level=info msg="kubectl --kubeconfig /dev/shm/876042171 -f - apply -n test-ops --dry-run" dir= execID=umTlO
time="2019-07-08T16:03:01Z" level=info msg="kubectl --kubeconfig /dev/shm/330626252 -f - apply -n test-ops --dry-run" dir= execID=zFIy3

**Comment by peterbosalliandercom** [comment]
Some failed some not

**Comment by alexec** [comment]
You should be able to set the log level to debug by starting the API server with `--loglevel debug`.

**Comment by alexec** [comment]
It looks like the error message is clipped off.

**Comment by alexec** [comment]
See https://github.com/argoproj/pkg/pull/4

**Comment by alexec** [comment]
There is no stderr in your error, so this indicates that kubectl didn't output any. Both stdout and the exit code would be useful. I wonder if this is a timeout we're not logging.

**Comment by peterbosalliandercom** [comment]
debug log does not tell a story:

time="2019-07-09T08:07:04Z" level=debug msg="Retrieved lived manifests" application=forza-tst-gitops
time="2019-07-09T08:07:04Z" level=debug msg="built managed objects list" application=forza-tst-gitops
time="2019-07-09T08:07:05Z" level=info msg="added resource Namespace/argo status: SyncFailed, message: " application=forza-tst-gitops
time="2019-07-09T08:07:05Z" level=debug msg="patch: {\"status\":{\"observedAt\":\"2019-07-09T08:07:03Z\",\"reconciledAt\":\"2019-07-09T08:07:03Z\",\"resources\":[{\"group\":
\"storage.k8s.io\",\"kind\":\"StorageClass\",\"name\":\"nfs\",\"status\":\"OutOfSync\",\"version\":\"v1\"},{\"group\":\"scheduling.k8s.io\",\"kind\":\"PriorityClass\",\"name
\":\"default\",\"status\":\"OutOfSync\",\"version\":\"v1beta1\"},{\"kind\":\"Namespace\",\"name\":\"kafka-int\",\"status\":\"OutOfSync\",\"version\":\"v1\"},{\"kind\":\"Name
space\",\"name\":\"kafka-acc\",\"status\":\"OutOfSync\",\"version\":\"v1\"},{\"kind\":\"Namespace\",\"name\":\"test-db-tunnel\",\"status\":\"OutOfSync\",\"version\":\"v1\"},
{\"kind\":\"Namespace\",\"name\":\"monitoring\",\"status\":\"OutOfSync\",\"version\":\"v1\"},{\"group\":\"storage.k8s.io\",\"kind\":\"StorageClass\",\"name\":\"gp2-encrypted
\",\"status\":\"OutOfSync\",\"version\":\"v1\"},{\"kind\":\"Namespace\",\"name\":\"test-acc\",\"status\":\"OutOfSync\",\"version\":\"v1\"},{\"kind\":\"Namespace\",\"name\":\
"test-int\",\"status\":\"OutOfSync\",\"version\":\"v1\"},{\"kind\":\"Namespace\",\"name\":\"kafka-prd\",\"status\":\"OutOfSync\",\"version\":\"v1\"},{\"kind\":\"Namespace\",
\"name\":\"test-ops\",\"status\":\"OutOfSync\",\"version\":\"v1\"},{\"group\":\"storage.k8s.io\",\"kind\":\"StorageClass\",\"name\":\"gp2\",\"status\":\"OutOfSync\",\"versio
n\":\"v1\"},{\"group\":\"scheduling.k8s.io\",\"kind\":\"PriorityClass\",\"name\":\"prod\",\"status\":\"OutOfSync\",\"version\":\"v1beta1\"},{\"kind\":\"Namespace\",\"name\":
\"analytics\",\"status\":\"OutOfSync\",\"version\":\"v1\"},{\"kind\":\"Namespace\",\"name\":\"argo\",\"status\":\"OutOfSync\",\"version\":\"v1\"},{\"kind\":\"Namespace\",\"n
ame\":\"default\",\"status\":\"OutOfSync\",\"version\":\"v1\"},{\"kind\":\"Namespace\",\"name\":\"keycloak\",\"status\":\"OutOfSync\",\"version\":\"v1\"},{\"kind\":\"Namespa
ce\",\"name\":\"kube-system\",\"status\":\"OutOfSync\",\"version\":\"v1\"},{\"group\":\"storage.k8s.io\",\"kind\":\"StorageClass\",\"name\":\"st1-encrypted\",\"status\":\"Ou
tOfSync\",\"version\":\"v1\"},{\"group\":\"scheduling.k8s.io\",\"kind\":\"PriorityClass\",\"name\":\"ops\",\"status\":\"OutOfSync\",\"version\":\"v1beta1\"},{\"kind\":\"Name
space\",\"name\":\"test-dev\",\"status\":\"OutOfSync\",\"version\":\"v1\"},{\"kind\":\"Namespace\",\"name\":\"test-monitoring\",\"status\":\"OutOfSync\",\"version\":\"v1\"},
{\"kind\":\"Namespace\",\"name\":\"helm\",\"status\":\"OutOfSync\",\"version\":\"v1\"},{\"kind\":\"Namespace\",\"name\":\"jenkins\",\"status\":\"OutOfSync\",\"version\":\"v1
\"},{\"kind\":\"Namespace\",\"name\":\"kube-public\",\"status\":\"OutOfSync\",\"version\":\"v1\"},{\"kind\":\"Namespace\",\"name\":\"istio-system\",\"status\":\"OutOfSync\",
\"version\":\"v1\"},{\"kind\":\"ServiceAccount\",\"name\":\"forza-tst-gitops-jenkins-inpai\",\"namespace\":\"test-ops\",\"status\":\"OutOfSync\",\"version\":\"v1\"},{\"health\":{\"status\":\"Healthy\"},\"kind\":\"Service\",\"name\":\"forza-tst-gitops-jenkins-inpai\",\"namespace\":\"test-ops\",\"status\":\"OutOfSync\",\"version\":\"v1\"},{\"kind\":\"ConfigMap\",\"name\":\"forza-tst-gitops-jenkins-inpai-tests\",\"namespace\":\"test-ops\",\"status\":\"OutOfSync\",\"version\":\"v1\"},{\"health\":{\"status\":\"Healthy\"},\"kind\":\"PersistentVolumeClaim\",\"name\":\"forza-tst-gitops-jenkins-inpai\",\"namespace\":\"test-ops\",\"status\":\"OutOfSync\",\"version\":\"v1\"},{\"health\":{\"status\":\"Healthy\"},\"kind\":\"Service\",\"name\":\"forza-tst-gitops-jenkins-inpai-agent\",\"namespace\":\"test-ops\",\"status\":\"OutOfSync\",\"version\":\"v1\"},{\"group\":\"rbac.authorization.k8s.io\",\"kind\":\"RoleBinding\",\"name\":\"forza-tst-gitops-jenkins-inpai-schedule-agents\",\"namespace\":\"test-ops\",\"status\":\"OutOfSync\",\"version\":\"v1\"},{\"kind\":\"ConfigMap\",\"name\":\"forza-tst-gitops-jenkins-inpai\",\"namespace\":\"test-ops\",\"status\":\"OutOfSync\",\"version\":\"v1\"},{\"group\":\"rbac.authorization.k8s.io\",\"kind\":\"Role\",\"name\":\"forza-tst-gitops-jenkins-inpai-schedule-agents\",\"namespace\":\"test-ops\",\"status\":\"OutOfSync\",\"version\":\"v1\"},{\"kind\":\"Secret\",\"name\":\"forza-tst-gitops-jenkins-inpai\",\"namespace\":\"test-ops\",\"status\":\"OutOfSync\",\"version\":\"v1\"},{\"group\":\"apps\",\"health\":{\"status\":\"Healthy\"},\"kind\":\"Deployment\",\"name\":\"forza-tst-gitops-jenkins-inpai\",\"namespace\":\"test-ops\",\"status\":\"OutOfSync\",\"version\":\"v1\"}]}}" application=forza-tst-gitops
time="2019-07-09T08:07:05Z" level=info msg="Update successful" application=forza-tst-gitops

**Comment by peterbosalliandercom** [comment]
It looks like there is something going wrong in the controller, I have changed the controller settings and after that suddenly the syncfailed resources dissappear. I changed these settings: 
```
       - argocd-application-controller
        - --status-processors
        - "20"
        - --operation-processors
        - "10"
        - --repo-server-timeout-seconds
        - "180"
```
to this
```
       - argocd-application-controller
        - --status-processors
        - "30"
        - --operation-processors
        - "30"

```

The repo server timeout we used (180) in order to prevent context deadline exceeded. 

I really don't know why this happens. any clues?

**Comment by alexec** [comment]
Did you see oomkilled for the repo server? I'm not clear.

**Comment by alexmt** [comment]
@peterbosalliandercom , We just realized that most likely `kubectl apply` was just OOM killed. This explains why `apply` fails with no error message. This also explains why the problem got resolved after you added memory. 

Right now it is unclear what happened because we don't log `kubectl apply ` return code. This is fixed by https://github.com/argoproj/argo-cd/pull/1891 . In v1.2 exit code should be visible in UI.

I would recommend to add `--repo-server-timeout-seconds` back and keep the memory.
